src/balance-fetcher.ts

```diff
@@ -19,6 +19,42 @@ export class BalanceFetcher {
     public readonly provider: ethers.Provider,
   ) {}
 
+  /**
+   * Creates balance data structure for a distributor with metadata and balances.
+   * @private
+   */
+  private createBalanceData(
+    address: string,
+    existingData: BalanceData | undefined,
+    newBalances: Record<string, string>,
+    allFetches: Array<{ address: string; date: string; block: number }>,
+  ): BalanceData {
+    const balanceData: BalanceData = {
+      metadata: {
+        chain_id: CHAIN_IDS.ARBITRUM_NOVA,
```

Comment:
if there is existing balance data then the chain id should already be set. Also, we never want to use arbnova as the chain id - always we want to get it from the provider.